lightning/src/ln/channelmanager.rs

```diff
@@ -1980,6 +2013,15 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		let onion_packet = onion_utils::construct_onion_packet(onion_payloads, onion_keys, prng_seed, payment_hash);
 
 		let _persistence_guard = PersistenceNotifierGuard::notify_on_drop(&self.total_consistency_lock, &self.persistence_notifier);
+		let mut pending_outbounds = self.pending_outbound_payments.lock().unwrap();
```

Comment:
This changes the lock ordering between `pending_outbound_payments` and `channel_state `. I noticed that in `write` we grab `channel_state` first. I haven't exhaustedly looked across method boundaries to see our ordering usage, but I wonder if something similar is what has been causing `background_processor_test` to hang.